core/src/main/java/org/infinispan/conflict/impl/DefaultConflictManager.java

```diff
@@ -0,0 +1,416 @@
+            rpcFuture = null;
+            keyOwners.clear();
+            if (rpcFuture.cancel(false)) {
+               retryQueue.add(this);
+
+               if (trace) log.tracef("Cancelling %s for nodes %s. New write owners %s", this, keyOwners, latestOwners);
+            }
+         }
+      }
+
+      void start() {
+         // TODO figure out if we can just use the CacheTopology.getPendingCh with segment id for a key instead of creating a mock LocalizedCacheTopology
+         // cacheTopology.getSegment(key)
+         //  cacheTopology.getPendingCH().locateOwnersForSegment();
+         // It should be fine, as keyPartitoner should not change during runtime, so we're basically using a non stored version of what is
+         // in a localized topology
+         keyOwners = installedTopology.getWriteOwners(key);
+
+         if (trace) log.tracef("Attempting %s from owners %s", this, keyOwners);
+
+         final Map<Address, InternalCacheValue<V>> versionsMap = new HashMap<>();
+         if (keyOwners.contains(localAddress)) {
+            GetCacheEntryCommand cmd = commandsFactory.buildGetCacheEntryCommand(key, localFlags);
+            InvocationContext ctx = invocationContextFactory.createNonTxInvocationContext();
+            InternalCacheEntry<K, V> internalCacheEntry = (InternalCacheEntry<K, V>) interceptorChain.invoke(ctx, cmd);
+            InternalCacheValue<V> icv = internalCacheEntry == null ? null : internalCacheEntry.toInternalCacheValue();
+            versionsMap.put(localAddress, icv);
+         }
+
+         ClusteredGetCommand cmd = commandsFactory.buildClusteredGetCommand(key, FlagBitSets.SKIP_OWNERSHIP_CHECK);
+         rpcFuture = rpcManager.invokeRemotelyAsync(keyOwners, cmd, rpcOptions);
+         rpcFuture.whenComplete((responseMap, exception) -> {
+            if (trace) log.tracef("%s received responseMap %s, exception %s", this, responseMap, exception);
+
+            if (exception != null) {
+               String msg = String.format("%s encountered when attempting '%s' on cache '%s'", exception.getCause(), this, cacheName);
+               completableFuture.completeExceptionally(new CacheException(msg, exception.getCause()));
+               return;
+            }
+
+            for (Map.Entry<Address, Response> entry : responseMap.entrySet()) {
+               if (trace) log.tracef("%s received response %s from %s", this, entry.getValue(), entry.getKey());
+               if (entry.getValue() instanceof SuccessfulResponse) {
+                  SuccessfulResponse response = (SuccessfulResponse) entry.getValue();
+                  Object rspVal = response.getResponseValue();
+                  versionsMap.put(entry.getKey(), (InternalCacheValue<V>) rspVal);
+               } else {
+                  completableFuture.completeExceptionally(new CacheException(String.format("Unable to retrieve key %s from %s: %s", key, entry.getKey(), entry.getValue())));
```

Comment:
non-successful response isn't the end of the world, you can get `UnsureResponse` when the target does not own given entry anymore, or `CacheNotFoundResponse` when the node was suspected...